547.py

```diff
@@ -0,0 +1,46 @@
+# This problem is to find number of connected component in an un-directed graph.
+
+# BFS Iterative Solution
+class Solution:
+    def findCircleNum(self, isConnected: List[List[int]]) -> int:
+        N = len(isConnected)
+        visited = [0] * N
+        queue = deque()
+        count = 0
+
+        for i in range(N):
+            if not visited[i]:
+                queue.append(i)
+
+                while queue:
+                    node = queue.popleft()
+                    visited[node] = 1
+
+                    for j in range(N):
+                        if isConnected[node][j] == 1 and visited[j] == 0:
```

Comment:
Maybe for this particular problem it happens to be right. But if you use this template to solve [LC-200](https://leetcode.com/problems/number-of-islands/) you would probably get a TLE due to `j` being added to `queue` repeatedly.